packages/bitcoin-wallet-provider/lib/BitcoinWalletProvider.ts

```diff
@@ -518,17 +528,53 @@ export default class BitcoinWalletProvider
     return { hex: txb.build().toHex(), fee };
   }
 
+  async quickGetAddresses(
+    startingIndex = 0,
+    numAddresses = 1,
+    change = false,
+  ): Promise<Address[]> {
+    if (numAddresses < 1) {
+      throw new Error('You must return at least one address');
+    }
+
+    const addresses = [];
+    const lastIndex = startingIndex + numAddresses;
+    const changeVal = change ? '1' : '0';
+
+    const originalProvider = this.client.getProviderForMethod('getAddresses');
```

Comment:
Would be good to mention here why we're grabbing the original provider, and why we're using `getDerivationPathAddressFn` because it's faster than calling `getMethod`